tests/Doctrine/Tests/DBAL/Functional/Driver/SQLAnywhere/StatementTest.php

```diff
@@ -50,4 +51,54 @@ public function testPersistentStatement()
         self::assertTrue($prepStmt->execute(),' Statement persistent failed');
     }
 
+    /**
+     * Check Issue #2991
+     * @throws \Doctrine\DBAL\ConnectionException
+     * @throws \Doctrine\DBAL\DBALException
+     * @throws \Exception
+     */
+    public function testBindParams()
```

Comment:
The case itself looks rather generic and not depending on a specific platform. Could it be moved to `Doctrine\Tests\DBAL\Functional\StatementTest`?

Besides that, the test case above already contains a tests which covers the `prepare()` → `bindValue()` → `execute()` flow (see `testReuseStatementWithReboundValue()`). Is this case somehow different from that one?

Reply:
ok, I moved it. Maybe another driver has the same problem, like sasql and furthermore oci8.
testReuseStatementWithReboundValue() can not reproduced the error. I think, we need a string-parameter.